invisible_cities/reco/deconv_functions_test.py

```diff
@@ -140,27 +141,51 @@ def test_interpolate_signal():
     assert np.allclose(grid             , sorted(set(inter_position[1])))
 
 
-def test_deconvolution_input(data_hdst, data_hdst_deconvolved):
+@fixture(scope="session")
+def new_grid_1mm():
+    bin_sizes = (1., 1.)
+    det_db    = DataSiPM('new', 0)
+    det_grid  = [ np.arange( det_db[var].min() - bs/2
+                           , det_db[var].max() + bs/2 + np.finfo(np.float32).eps, bs)
+                  for var, bs in zip("X Y".split(), bin_sizes)]
+    return det_grid
+
+
+def test_deconvolution_input(data_hdst, data_hdst_deconvolved, new_grid_1mm):
     ref_interpolation = np.load(data_hdst_deconvolved)
 
     hdst   = load_dst(data_hdst, 'RECO', 'Events')
     h      = hdst[(hdst.event == 3021916) & (hdst.npeak == 0)]
     h      = h.groupby(['X', 'Y']).Q.sum().reset_index()
     h      = h[h.Q > 40]
 
-    det_db   = DataSiPM('new', 0)
-    det_grid = [np.arange(det_db[var].min() + bs/2, det_db[var].max() - bs/2 + np.finfo(np.float32).eps, bs)
-               for var, bs in zip(['X', 'Y'], [1., 1.])]
-    interpolator = deconvolution_input([10., 10.], det_grid, InterpolationMethod.cubic)
+    interpolator = deconvolution_input([10., 10.], new_grid_1mm, InterpolationMethod.cubic)
     inter        = interpolator((h.X, h.Y), h.Q)
 
     assert np.allclose(ref_interpolation['e_inter'], inter[0])
     assert np.allclose(ref_interpolation['x_inter'], inter[1][0])
     assert np.allclose(ref_interpolation['y_inter'], inter[1][1])
 
 
+@mark.parametrize("interp_method", InterpolationMethod)
+def test_deconvolution_input_interpolation_method(data_hdst, new_grid_1mm, interp_method):
+    """
+    Check that it runs with all interpolation methods
+    """
+    hdst = load_dst(data_hdst, 'RECO', 'Events')
+    hdst = hdst[(hdst.event == 3021916) & (hdst.npeak == 0) & (hdst.Q>40)]
+
+    deconvolve = deconvolution_input([10., 10.], new_grid_1mm, interp_method)
+    output     = deconvolve((hdst.X, hdst.Y), hdst.Q)
```

Comment:
This is really nitpicky, but `deconvolve()` is the name of a function within `deconv_functions.py`, and differs from your naming in `test_deconvolution_input` of `interpolator()`, a function that achieves the same task.